### PR TITLE
dead link checker

### PR DESCRIPTION
**Pull request by stelcheck**
Also updated the package-lock.json file since it seems like
it was pretty much out of date.

### CODE REVIEW

**Comment by ronkorving** [comment]
Maybe we can add mdlint to this?

**Comment by stelcheck** [comment]
We could, but I would perhaps indeed let this go into another PR.

**Comment by stelcheck** [comment]
PR was updated to solve merge conflicts.

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "test:outdated": "npm outdated --color always --silent || true",
     "test:security": "retire && npm run test:outdated",
     "test:lint": "eslint --ext .js,.json .",
+    "test:docs": "blcl ./docs -ro",
```
Comment (ronkorving) [inline package.json]: We don't run this on `npm test`?

Comment (stelcheck) [reply package.json]: We can't. Reason is that sometimes you want to link to, say, the API doc, but it hasn't been published yet.

Comment (ronkorving) [reply package.json]: Maybe run it in a prepublish hook or something?

Comment (stelcheck) [reply package.json]: We could... just not sure if there might be any other caveats I am not thinking of. 

I'd say we could add this to pre-publish at a later time, once we've used that tool a bit more by hand.

**Pull request merged**